packages/libp2p/src/transport-manager.ts

```diff
@@ -262,27 +257,27 @@ export class DefaultTransportManager implements TransportManager, Startable {
    * If a transport has any running listeners, they will be closed.
    */
   async remove (key: string): Promise<void> {
-    const listeners = this.listeners.get(key) ?? []
-    log.trace('removing transport %s', key)
-
     // Close any running listeners
-    const tasks = []
-    log.trace('closing listeners for %s', key)
-    while (listeners.length > 0) {
-      const listener = listeners.pop()
-
-      if (listener == null) {
-        continue
-      }
-
-      tasks.push(listener.close())
+    for (const listener of this.listeners.get(key)?.values() ?? []) {
+      await listener.close()
     }
```

Comment:
Do we want to map over these so we can wait for multiple listener.close() in parallel instead;  Similar to how we’re doing in .stop()